### PR TITLE
Makes Playing Lobby Music A Config

### PR DESCRIPTION
**Pull request by san7890**
## About The Pull Request

Hey there,

Apparently some people don't like listening to the soulful lobby music we have to offer. How unfortunate. This adds a config flag to disable said title screen music. I've heard people who like debugging a lot don't want to get their ears-bleeding via flipflap, but I find it hard to agree.

I will not lie to you, my heart honestly sunk to the bowels of my gut when I finally got this to work. If you toggle this, you are ludicrous.

## Why It's Good For The Game

Some people don't like being bugged by the title screen music (even though it's such a helpful reminder for when your compile successfully goes through while you're scrolling Twitter or something). You can enable this config on your local fork whenever it gets annoying and you aren't logged into BYOND since your preferences aren't set. If you are a server host and do not like having our free lobby music play, you can also do this. Just one flag to set (and to never commit).

## Changelog

:cl:
config: Hey, server operators! Title music playing at the lobby screen is now DISABLED by default in the config settings (game_options.txt). If you are not hearing any title music, be sure to adjust your config. If you're a player reading this and are sorely missing out on those soulful tunes, please notify your server's administration team of this change so they can diagnose it properly from there.
/:cl:

### CODE REVIEW

File: code/controllers/configuration/entries/game_options.dm [hunk 1]
```diff
@@ -391,7 +391,10 @@
 
 /datum/config_entry/flag/native_fov
 
+/datum/config_entry/flag/allow_title_music
```
Comment (Mothblocks) [inline code/controllers/configuration/entries/game_options.dm:394]: this is gonna default to false bro

Comment (san7890) [reply code/controllers/configuration/entries/game_options.dm]: whoops, i just decided to rewrite it

**Comment by Iamgoofball** [comment]
Why? There's already a preference to turn it off.

**Comment by MrStonedOne** [comment]
> @MrStonedOne do you not switch branches in place?

No. i give each branch a clean starting folder because i had phantom bugs once that didn't show up in a clean folder and it lost me 5 hours of time so i said never again.

File: config/game_options.txt [hunk 2]
```diff
@@ -466,3 +466,6 @@ MAXFINE 2000
 
 ## Whether native FoV is enabled for all people.
 #NATIVE_FOV
+
+## Uncomment if you wish to disable title music from playing at the lobby screen.
+#DISALLOW_TITLE_MUSIC
```
[the remark this answers is not in the document]
Comment (SuperNovaa41) [reply config/game_options.txt:471]: ```Comment if you wish to enable```

**Comment by san7890** [comment]
@MrStonedOne Changes should be applied, works on my local machine.

**Comment by Mothblocks** [comment]
ugh i need to get local overrides working

File: code/game/sound.dm [hunk 3]
```diff
@@ -199,7 +199,7 @@ distance_multiplier - Can be used to multiply the distance at which the sound is
 	set waitfor = FALSE
 	UNTIL(SSticker.login_music) //wait for SSticker init to set the login music
 
-	if(prefs && (prefs.toggles & SOUND_LOBBY))
+	if(prefs && (prefs.toggles & SOUND_LOBBY) && (!CONFIG_GET(flag/disallow_title_music)))
```
Comment (Kylerace) [inline code/game/sound.dm:202]:
```suggestion
	if(prefs && (prefs.toggles & SOUND_LOBBY) && !CONFIG_GET(flag/disallow_title_music))
```

**Pull request merged**